jmclient/jmclient/wallet_service.py

```diff
@@ -74,18 +75,19 @@ def __init__(self, wallet):
         # Dicts of registered callbacks, by type
         # and then by txinfo, for events
         # on transactions.
-        self.callbacks = {}
-        self.callbacks["all"] = []
-        self.callbacks["unconfirmed"] = {}
-        self.callbacks["confirmed"] = {}
+        self.callbacks = {
+            "all": [],  # note: list, not dict
+            "unconfirmed": {},
+            "confirmed": {},
+        }
 
         self.restart_callback = None
 
         # transactions we are actively monitoring,
         # i.e. they are not new but we want to track:
-        self.active_txids = []
+        self.active_txs = {}
```

Comment:
Why did you change the name to `txs` from `txids`? They are actually txids, right?

Reply:
The keys are txids. The values are the deserialized transactions as returned by `get_deser_from_gettransaction`.

Reply:
I see in the other pending merge request, set() was used for both.
Does this make sense to change here too to         `self.active_txs = set()`
?

https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/1042/commits/0bd390d3d5fe6979bba9808aec1f0e6aa325b791#diff-fdd873116e791aacd2db174034fa598cca422b8ef835c2e4a1a630861107f471

Reply:
`{}` is a dictionary. Uniqueness of keys is enforced, just as a set enforces uniqueness of elements.

Reply:
Yes, that all seems entirely reasonable. Thanks for the explanation.